src/sets.jl

```diff
@@ -46,6 +46,27 @@ abstract type AbstractVectorSet <: AbstractSet end
 
 dimension(s::AbstractVectorSet) = s.dimension # .dimension field is conventional, overwrite this method if not applicable
 
+"""
+    set_dot(x::Vector, y::Vector, set::AbstractVectorSet)
+
+Return the scalar product between a vector `x` of the set `set` and a vector
+`y` of the dual of the set `s`.
+"""
+function set_dot(x::Vector, y::Vector, set::AbstractVectorSet)
```

Comment:
If someone defines a new set with a non-trivial inner product, he is forced to define `set_dot` and `dot_coefficients` otherwise all codes using these functions fails. As this is mandatory, this is part of the API, isn't it ?

Reply:
Is there code using these functions outside of MOIU? The `get_fallback` should be opt-in by the solver and not automatic.